hedera-node/hedera-mono-service/src/test/java/com/hedera/services/contracts/execution/LivePricesSourceTest.java

```diff
@@ -112,4 +126,30 @@ private void givenCollabsWithMultiplier(final long multiplier) {
         given(feeMultiplierSource.currentMultiplier(accessor)).willReturn(multiplier);
         given(txnCtx.accessor()).willReturn(accessor);
     }
+
+    @Test
+    void getsCurrentGasPriceInTinyCents() {
+        given(usagePrices.defaultPricesGiven(ContractCall, timeNow)).willReturn(providerPrices);
+
+        ToLongFunction<FeeComponents> resourcePriceFn = FeeComponents::getGas;
+        final var expected = resourcePriceFn.applyAsLong(providerPrices.getServicedata()) / 1000;
+        assertEquals(expected, subject.currentGasPriceInTinycents(timeNow, ContractCall));
+    }
+
+    @Test
+    void estimatesFutureGasPriceInTinybars() {
+        given(exchange.rate(timeNow)).willReturn(currentRate);
+        given(usagePrices.defaultPricesGiven(CryptoCreate, timeNow))
+                .willReturn(defaultCurrentPrices);
+
+        // and:
+        long expected =
+                getTinybarsFromTinyCents(currentRate, mockFees.getGas() / FEE_DIVISOR_FACTOR);
+
+        // when:
+        long actual = subject.estimatedGasPrice(CryptoCreate, timeNow);
+
+        // then:
+        assertEquals(expected, actual);
```

Comment:
There shouldn't be any new tests for `LivePricesSource` since no changes are required for this class.